frigate/prometheus_exporter.py

```diff
@@ -0,0 +1,219 @@
+import json
+import re
+from urllib.request import urlopen
+from prometheus_client import CollectorRegistry
+from prometheus_client.metrics_core import GaugeMetricFamily, InfoMetricFamily
+
+
+def setupRegistry():
+    myregistry = CollectorRegistry()
+    myregistry.register(CustomCollector())
+    return myregistry
+
+
+def add_metric(metric, label, data, key, multiplier=1.0):
+    try:
+        string = str(data[key])
+        value = float(re.findall(r"\d+", string)[0])
+        metric.add_metric([label], value * multiplier)
+    except (KeyError, TypeError, IndexError):
+        pass
+
+
+class CustomCollector:
+    def __init__(self):
+        self.stats_url = "http://localhost:5000/api/stats"
+
+    def collect(self):
+        data = json.loads(urlopen(self.stats_url).read())
+
+        # camera stats
+        ffmpeg_pid = GaugeMetricFamily(
+            "frigate_ffmpeg_pid", "PID for ffmpeg process", labels=["camera"]
+        )
```

Comment:
```suggestion
        ffmpeg_up = GaugeMetricFamily(
            "frigate_ffmpeg_up", "Whether the ffmpeg process for a camera is up", labels=["camera"]
        )
```

Reply:
If it restarts and doesn't stay down, that is great. But such an event is ultimately pretty uninteresting from a monitoring perspective where the service availability is of interest.

Reply:
> Edit: Oh, but it may be entirely unsuitable for noticing the process is actually down in that case.

Right, that is my point. Once frigate has started and the process is up, this will always show as the process being up even in the case where it continues to crash

Reply:
We are actually using the camera_fps for this case. The camera_fps will be set to 0 when the ffmpeg process has crashed.

I am not sure if the PID is useful for this particular application

Reply:
The process is a `multiprocessing.Process`, in which case `is_alive()` would be a better option.